tiff-2-png-converter.py

```diff
@@ -1,197 +1,358 @@
 import argparse
 import sys
-import os 
+import os
 import cv2
 import numpy as np
 import imagesize
 import math
-import tifffile  as tiff
+import tifffile as tiff
 import numpngw
-from progress_bar import *
-
-verbose = False
-
-
-
-
-def convert_tiff_directory_action(source_path, dest_path):
-    list_files,width, heigth = list_folder_files(source_path)
-    num_images = len(list_files)
-    image_r = [None] * num_images
-    image_g = [None] * num_images
-    image_b = [None] * num_images
-    images = [None] * num_images
-    bits_per_sample = ""
-    printProgressBar(0, num_images, prefix = 'grouping channels:', suffix = 'Complete', length = 50)
-    for i in range(num_images):
-        file_full_path = list_files[i]
-        img = tiff.imread(list_files[i])
-        if bits_per_sample=="":
-            bits_per_sample = img.dtype
-        elif bits_per_sample!=img.dtype:
-            print("image " + file_full_path + "has a different sample type "+bits_per_sample+" expected: "+bits_per_sample)
-            sys.exit("Check images bits per sample tag")
-        name, extension = os.path.splitext(file_full_path)
-        if "ch1" in name:
-             if verbose:
-                print("Load Image " + str(i) + " Channel 1 - "+ file_full_path )
-             image_r[i] = img
-        elif "ch2" in name:
-             if verbose:
-                print("Load Image " + str(i) + " Channel 2 - "+ file_full_path )
-             image_g[i] = img
-        elif "ch3" in name:
-             if verbose:
-                print("Load Image " + str(i) + " Channel 3 - "+ file_full_path )
-             image_b[i] = img
+from progress_bar import print_progress_bar
+from image_process_helper import *
+from scipy.interpolate import interp1d
+import re
+
+import typer
+
+
+class TiffProcessor:
+    verbose = False
+    equalize_histogram = False
+    equalize_histogram_method = None
+    gamma = False
+    gamma_val = 1
+    num_imgs = None
+    source = None
+    dest = None
+
+    @staticmethod
+    def add_arguments(parser):
+        parser.add_argument(
+            "-s",
+            "--source",
+            required=True,
+            help="Path to directory where the image sequence is located",
+        )
+        parser.add_argument(
+            "-d",
+            "--dest",
+            required=True,
+            help="Path to the directory and filename of the result merged image (png by default) will be saved. ",
+        )
+        parser.add_argument(
+            "--verbose", help="Generate messages ouput ", action="store_true"
+        )
+
+        parser.add_argument("--num_imgs", help="Generate messages ouput ")
+
+        parser.add_argument(
+            "--equalize1",
+            help="equilize histogram in volume slices using opencv functionality",
+            action="store_true",
+        )
+
+        parser.add_argument(
+            "--equalize2",
```

Comment:
these names should be more descriptive.  or perhaps make the type of equalizer a value passed to an `--equalize` flag

Reply:
1. All the functions in the image_process_helper file are being used. In this case, Is it necessary to be specific ?
`from image_process_helper import gamma_correction,equalize_image_histogram_opencv,equalize_image_histogram_custom`
vs
`from image_process_helper import *`

2. Let me rework this in the typer migration. I have to change the command args either way. I'll put it in the issue.

Reply:
Part of the reason this pattern is discouraged is for provenance tracking.  So maybe do `import image_process_helper as iph` then prepend your function calls with `iph.`?

Reply:
Good call! I'm adding it to the latest changes